src/coreComponents/mesh/MeshLevel.cpp

```diff
@@ -202,6 +202,10 @@ MeshLevel::MeshLevel( string const & name,
       allNodes.insert( a );
     }
 
+    SortedArray< localIndex > & zposNodes  = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "zpos" ) ).reference();
+    zposNodes = source.m_nodeManager->getSet("zpos");
+    m_faceManager->buildSets( *m_nodeManager );
```

Comment:
In the simulations that use the internal mesh generator, `zpos` is always defined by default. However, in the simulations that use an imported mesh, `zpos` is not defined by default. So maybe here, we should create a set `zpos` in the high-order mesh level only it is already defined in the base mesh?